controllers/admin/AdminCarriersController.php

```diff
@@ -642,13 +642,13 @@ public function changeZones($id)
      */
     public function getList($id_lang, $order_by = null, $order_way = null, $start = 0, $limit = null, $id_lang_shop = false)
     {
-        parent::getList($id_lang, $order_by, $order_way, $start, $limit, $id_lang_shop);
+        // Replace "0" by the name of the shop directly in SQL query (allowing sort without errors)
+        $this->_select = 'CASE WHEN name = "0" '
+            . 'THEN "' . pSQL(Carrier::getCarrierNameFromShopName()) . '" '
+            . 'ELSE name '
+            . 'END AS name';
```

Comment:
Why not use if?
```suggestion
        $this->_select .= sprintf(
            ', IF(name = "0", "%s", name) AS name',
            pSQL(Carrier::getCarrierNameFromShopName())
        );
```

Reply:
Thanks for the review. (And if have forgot the  `IF`).